### PR TITLE
Wayland fixes

### PR DESCRIPTION
**Pull request by pkerling**
The wl_registry global_remove handler was set to NULL, which is not allowed and will crash the program if it is invoked.

### CODE REVIEW

**Comment by pkerling** [comment]
Note: The other commit is not directly related, I can make two PRs out of this if needed.

**Comment by chivakker** [comment]
@pkerling would you mind filing a bug with a description?  Then add Fixes to your commit message.

File: va/wayland/va_wayland_drm.c [hunk 1]
```diff
@@ -239,46 +226,80 @@ va_wayland_drm_create(VADisplayContextP pDisplayContext)
...
     if (!wl_drm_ctx->is_authenticated) {
         va_wayland_error("Wayland compositor did not respond to DRM authentication");
-        return false;
+        goto end;
+    }
+
+    result = true;
+
+end:
+    if (wrapped_display) {
```
Comment (xhaihao) [inline va/wayland/va_wayland_drm.c]: It is possible wrapped_display is not initialized.   please set wrapped_display to NULL at the beginning.

Comment (pkerling) [inline va/wayland/va_wayland_drm.c]: fixed, thanks!

**Comment by pkerling** [comment]
@chivakker done

**Comment by chivakker** [comment]
The patches need rebase and the removal of the last Merge branch commit.

**Comment by pkerling** [comment]
@chivakker the merge commit is by @xhaihao, not sure what I should do here?

**Comment by uartie** [comment]
The merge commit is just a side-effect of rebase via GitHub UI.  @xhaihao will likely deal with it when he's ready to push it to master.

**Comment by xhaihao** [comment]
Don't worry about the merge commit.  We enabled 'rebase and merge' only for pushing the code, so the merge commit will disappear after merging the code.

**Pull request merged**